File: backend/ms_email/src/main/java/wevioo/tn/ms_email/entities/EmailTemplate.java
package wevioo.tn.ms_email.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class EmailTemplate implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String language;
    private State state;
    @ElementCollection(fetch = FetchType.EAGER)
    private List<Long> userFavoriteEmails = new ArrayList<>();
    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER , orphanRemoval = true)
    @JoinColumn(name = "template_body_id", referencedColumnName = "id")
    private TemplateBody templateBody;
}

File: frontend/src/models/email/EmailTemplate.ts
import { SmsTemplate } from "../sms/SmsTemplate";
import { TemplateBody } from "./TemplateBody";

export interface EmailTemplate{
    id?:number,
    name: string,
    language:string,
    state?:string,
    templateBody:TemplateBody
    userFavoriteEmails?: number[];
}

export function isEmailTemplate(template: EmailTemplate | SmsTemplate): template is EmailTemplate {
    return (template as EmailTemplate).state !== undefined;
  }